### DISCUSSION TITLE
fixed

### DISCUSSION THREAD

**Discussion by MorrisNijland3** [pyinstaller/pyinstaller#8382] [Help] [answered]
Fixed

**Comment by bwoodsend** [comment]
Its working a month ago doesn't really tell us anything. We need the error message and whatever code is used to locate and process these JSON files and the details of how you build your application.

**Reply by MorrisNijland3** [reply]
My bad forgot that, well the code is basically nothing more than reading the json file and displaying 1 row of the file in excel.
I thought maybe someone knew a bug with wine and pyinstaller.

**Reply by rokm** [reply]
Assuming you are building with console enabled, if you add a couple of `print` statements to different places in your program script, do you see (any of) them displayed when you try to run the program?

**Reply by bwoodsend** [reply]
That's not code or an error message. We still have nothing to go on here.

**Reply by MorrisNijland3** [reply]
> Assuming you are building with console enabled, if you add a couple of `print` statements to different places in your program script, do you see (any of) them displayed when you try to run the program?

Yes, at every place I see the print messages. 
The program uses xlsxwriter and prints the useful information from the json file on the excel sheet.
If I make the executable file for macOS, is runs perfectly fine, but only when it is made with wine for Windows, it runs ok, but makes no excel sheet.

**Reply by rokm** [reply]
> > Assuming you are building with console enabled, if you add a couple of `print` statements to different places in your program script, do you see (any of) them displayed when you try to run the program?
> 
> Yes, at every place I see the print messages. The program uses xlsxwriter and prints the useful information from the json file on the excel sheet. If I make the executable file for macOS, is runs perfectly fine, but only when it is made with wine for Windows, it runs ok, but makes no excel sheet.

If the program reaches the end (i.e., if you put a `print` at the end of the program and you see that message printed), then either it succeeded but the file was written in a different location than you expect (how are you specifying the output location?), or there was an exception but you swallowed it, so you don't see the error. Either way, we cannot help you without seeing the code.

**Reply by MorrisNijland3** [reply]
> > > Assuming you are building with console enabled, if you add a couple of `print` statements to different places in your program script, do you see (any of) them displayed when you try to run the program?
> 
> > 
> 
> > Yes, at every place I see the print messages. The program uses xlsxwriter and prints the useful information from the json file on the excel sheet. If I make the executable file for macOS, is runs perfectly fine, but only when it is made with wine for Windows, it runs ok, but makes no excel sheet.
> 
> 
> 
> If the program reaches the end (i.e., if you put a `print` at the end of the program and you see that message printed), then either it succeeded but the file was written in a different location than you expect (how are you specifying the output location?), or there was an exception but you swallowed it, so you don't see the error. Either way, we cannot help you without seeing the code.




```

import xlsxwriter as xw
import json, os, sys
from datetime import datetime




def get_script_directory():
    if getattr(sys, 'frozen', False):  
        return os.path.dirname(sys.executable)
    else:
        return os.path.dirname(os.path.abspath(__file__))

script_dir = get_script_directory()

workbook_path = os.path.join(script_dir, 'taken.xlsx')
workbook = xw.Workbook(workbook_path)

filename = 'Taken.json'
json_path = os.path.join(script_dir, filename)

print("Attempting to open file at path:", json_path)  

if os.path.exists(json_path):
    with open(json_path) as file:
        reader = json.load(file)


workbook = xw.Workbook(os.path.join(get_script_directory(), 'taken.xlsx'))
if workbook:
    print('ok')

personen = {'b4d1b492-7032-463d-83af-aff06efa96ce' : 'Alec van der Schuit',
            '476a5379-0488-4d86-8fe1-891d7395f362' : 'Amber Schouten',
            '229783bd-5be2-4298-a0f1-1ce089b6e621' : 'Daan Bakker',
            '8d303b8f-4dad-442c-89b4-aa7c76936416' : 'Dionne Duijzer',
            '413d0ff7-8dcb-420a-ba12-951f36f7d67e' : 'Ewout Vet',
            '9d190cd0-e1e2-4cc1-97ec-1e0c27e7bfbc' : 'Jelmer de Vries',
            'f57b2dd5-443a-4306-a2e3-f14b44f0501a' : 'Julia van Kalken',
            '4282d6cb-daad-438d-b530-4e7bf1b23519' : 'Julia Klaver',
            '72222403-1d90-49d5-9da3-780a0b39fb8b' : 'Kay van Wijk',
            '5b092ce6-0f82-4b33-aad7-6e3cb6486950' : 'Leonie van der Park',
            '59549484-de76-4dac-90f5-1f09f644bed6' : 'Miel Tiebie',
            '5efa6a82-5942-4dd9-a54e-fa71365e8292' : 'Morris Nijland',
            '8f1c96b3-8224-48f7-b4c7-a7e74338dfc8' : 'Sabrina Schot',
            '99dba567-3892-4236-a695-2e8c0180c197' : 'Steijn van Buuren',
            '5c84a86f-c592-40f1-af8c-d8e7d108e37e' : 'Thijmen Buurs',
            'a1266dc0-cdf5-472c-9f17-2f851ebc2f76' : 'Tijn de Ruijter'
            }       # personen met hun teams id


for werknemer in personen:
    index = 0
    worksheet = workbook.add_worksheet(personen[werknemer])
    today = datetime.now()
    for read in reader:
        titel = read["Title"]
        plan = read['Plan']
        assigned_to = read["AssignedToUserId"]
        due = read["Due"][:10]
        voortgang = read["Voortgang"]
        persoon_ = []

        reader1 = json.loads(assigned_to)

        for werkn in reader1:
            persoon_.append(werkn)

        if werknemer in persoon_:
            if voortgang != '100':
                worksheet.write(0, index, titel)
                worksheet.write(1, index, plan)
                if due:
                    year = int(due[0:4])
                    month = int(due[5:7])
                    day = int(due[8:10])
                    due_till = f'{day}/{month}/{year} 23:59'
                    due_till = datetime.strptime(due_till, '%d/%m/%Y %H:%M')
                    if due_till < today:
                        worksheet.write(2, index, f'Te laat({due})')
                    else:
                        worksheet.write(2, index, due)
                        
                else:
                    worksheet.write(2, index, 'Geen datum')

                personen_i = 2
                for i in persoon_:
                    if i in personen:
                        worksheet.write(personen_i + 1, index, personen[i])
                        personen_i += 1
                index += 3

workbook.close()
print('done')

```


This is the code.
I'm sorry if it looks messy, just started with coding.

If I'm right it stores the file at the executed file location.

**Reply by bwoodsend** [reply]
That works for me under wine (albeit with a Linux host). I get a `taken.xlsx` next to the executable.

**Reply by MorrisNijland3** [reply]
> That works for me under wine (albeit with a Linux host). I get a `taken.xlsx` next to the executable.

Mhmm weird, with Mac and windows both work.
But when used with windows it won't.

**Reply by rokm** [reply]
In that case, can you archive both the `build` and `dist` directory, and upload the archive somewhere so I can look at it? Could be that we end up collecting a Wine DLL somewhere, and that causes problems on real Windows (I recall putting in some checks against that some time ago, but it could be from a different codepath).

**Reply by rokm** [reply, marked as answer]
> The first problem was not importing xlsxwriter, this problem is fixed if I'm right.

The problem seems to be that `xlsxwriter` is not packaged with the application:

```
>taken_overzicht.exe
Traceback (most recent call last):
  File "taken_overzicht.py", line 1, in <module>
ModuleNotFoundError: No module named 'xlsxwriter'
[2984] Failed to execute script 'taken_overzicht' due to unhandled exception!
```

You can aslo see

```
xlsxwriter MissingModule
imported by: taken_overzicht.py
```

in `build\taken_overzicht\xref-taken_overzicht.html`, which likely means that the module is missing from the python environment during the build.

**Reply by rokm** [reply]
> Which is weird, cause I explicitly used the environment with xlsxwriter and even used —hidden-imports.

If a module is not in the search path, no amount of hidden imports will help you...

Do you have multiple PyInstaller installations on your system? Are you running the one from this environment?

**Reply by rokm** [reply]
Also, if you use the latest PyInstaller release, it should print the search paths at the beginning of the build process. That should help you figure out what environment is actually in use.
